Type error because List is invariant

Version info:
```
mypy                    0.812
pytest-subprocess       1.1.0
```
Error:
```
main.py:34: error: Argument 1 to "register_subprocess" of "FakeProcess" has incompatible type "List[str]"; expected "Union[List[Union[str, Any]], Tuple[Union[str, Any], ...], str, Command]"
main.py:34: note: "List" is invariant -- see http://mypy.readthedocs.io/en/latest/common_issues.html#variance
main.py:34: note: Consider using "Sequence" instead, which is covariant
main.py:34: error: Argument "stdout" to "register_subprocess" of "FakeProcess" has incompatible type "List[str]"; expected "Union[str, bytes, List[Union[str, bytes]], Tuple[Union[str, bytes], ...], None]"
```
https://mypy-play.net/?mypy=latest&python=3.9&gist=89232479f9811be76ee54460a5f1a895

Fixable if you annotate the arguments as `List[Union[str, bytes]]`.

Does `register_subprocess` really require the ability to mutate the input list? If it only needs read-only access, then using `Sequence` would be helpful to avoid the unnecessary upcast.

--- a/changelog.d/other.ddf9f4ed.entry.yaml
+++ b/changelog.d/other.ddf9f4ed.entry.yaml
@@ -1,0 +1,8 @@
+git_email: github@aklajnert.pl
+git_user: Andrzej Klajnert
+message: Fix type annotations for `register_subprocess()`.
+os_user: andrzejk
+pr_ids:
+- '42'
+timestamp: 1620915915
+type: other

--- a/noxfile.py
+++ b/noxfile.py
@@ -16,7 +16,9 @@ def flake8(session):
 @nox.session
 def mypy(session):
     session.install("mypy")
+    session.run("mypy", "--version")
     session.run("mypy", "pytest_subprocess", "--config-file=setup.cfg")
+    session.run("mypy", "tests/typing.py", "--config-file=setup.cfg")
 
 
 @nox.session

--- a/pytest_subprocess/core.py
+++ b/pytest_subprocess/core.py
@@ -25,11 +25,11 @@ from .utils import Thread
 
 OPTIONAL_TEXT = Union[str, bytes, None]
 OPTIONAL_TEXT_OR_ITERABLE = Union[
-    str, bytes, None, List[Union[str, bytes]], Tuple[Union[str, bytes], ...],
+    str, bytes, None, Sequence[Union[str, bytes]], Tuple[Union[str, bytes], ...],
 ]
 BUFFER = Union[None, io.BytesIO, io.StringIO]
 ARGUMENT = Union[str, Any]
-COMMAND = Union[List[ARGUMENT], Tuple[ARGUMENT, ...], str, Command]
+COMMAND = Union[Sequence[ARGUMENT], Tuple[ARGUMENT, ...], str, Command]
 
 
 class PluginInternalError(Exception):
@@ -49,9 +49,7 @@ class FakePopen:
         self,
         command: Union[
             Union[bytes, str],
-            Sequence[
-                Union[str, bytes, "os.PathLike[str]", "os.PathLike[bytes]"]
-            ]
+            Sequence[Union[str, bytes, "os.PathLike[str]", "os.PathLike[bytes]"]],
         ],
         stdout: OPTIONAL_TEXT_OR_ITERABLE = None,
         stderr: OPTIONAL_TEXT_OR_ITERABLE = None,

--- a/pytest_subprocess/utils.py
+++ b/pytest_subprocess/utils.py
@@ -3,6 +3,7 @@ from typing import Any as AnyType
 from typing import Iterator
 from typing import List
 from typing import Optional
+from typing import Sequence
 from typing import Tuple
 from typing import Union
 
@@ -27,7 +28,7 @@ class Command:
     __slots__ = "command"
 
     def __init__(
-        self, command: Union[Tuple[ARGUMENT, ...], List[ARGUMENT], str],
+        self, command: Union[Tuple[ARGUMENT, ...], Sequence[ARGUMENT], str],
     ):
         if isinstance(command, str):
             command = tuple(command.split(" "))

--- a/tests/typing.py
+++ b/tests/typing.py
@@ -1,0 +1,9 @@
+"""Additional target for mypy type checking"""
+from pytest_subprocess import FakeProcess
+
+
+def test_typing() -> None:
+    fp = FakeProcess()
+    cmd = ["ls", "-l"]
+    output = ["some", "lines", "of", "output"]
+    fp.register_subprocess(cmd, stdout=output)

